sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -314,6 +314,27 @@ case class Divide(left: Expression, right: Expression) extends DivModLike {
   override def evalOperation(left: Any, right: Any): Any = div(left, right)
 }
 
+@ExpressionDescription(
+  usage = "a _FUNC_ b - Divides a by b.",
+  examples = """
+    Examples:
+      > SELECT 3 _FUNC_ 2;
+       1
+  """,
+  since = "3.0.0")
+case class IntegralDivide(left: Expression, right: Expression) extends DivModLike {
```

Comment:
Shall we add this to `FunctionRegistry`?

Reply:
I don't think so, please see the discussion at https://github.com/apache/spark/pull/14036#discussion_r69392061

Reply:
Ur, sorry, but why not? As @viirya suggested, without that, the description added here is not meaningless.
```scala
spark-sql> describe function 'div';
Function: div not found.
Time taken: 0.016 seconds, Fetched 1 row(s)
```

Also, Hive accepts that like the following.
```scala
0: jdbc:hive2://ctr-e138-1518143905142-429335> describe function div;
+----------------------------------------------------+
|                      tab_name                      |
+----------------------------------------------------+
| a div b - Divide a by b rounded to the long integer |
+----------------------------------------------------+

0: jdbc:hive2://ctr-e138-1518143905142-429335> select 3 / 2, 3 div 2, `/`(3,2), `div`(3,2);
+------+------+------+------+
| _c0  | _c1  | _c2  | _c3  |
+------+------+------+------+
| 1.5  | 1    | 1.5  | 1    |
+------+------+------+------+
1 row selected (0.288 seconds)
```

Reply:
@dongjoon-hyun because if we add it there, we can write: `select div(3, 2)`, which is not supported by Hive.

```
hive> select div(3, 2);
NoViableAltException(13@[])
	at org.apache.hadoop.hive.ql.parse.HiveParser_SelectClauseParser.selectClause(HiveParser_SelectClauseParser.java:964)
```

Reply:
@mgaido91 . I gave the example of Hive in the above. :)

Reply:
ah, sorry I missed the back-ticks. I am adding it, sorry. Thanks.